aries_cloudagent/wallet/routes.py

```diff
@@ -544,6 +568,31 @@ async def wallet_rotate_did_keypair(request: web.BaseRequest):
     return web.json_response({})
 
 
+def register_events(event_bus: EventBus):
+    """Subscribe to any events we need to support."""
+    event_bus.subscribe(EVENT_LISTENER_PATTERN, on_register_nym_event)
+
+
+async def on_register_nym_event(profile: Profile, event: Event):
+    """Handle any events we need to support."""
+
+    # after the nym record is written, promote to wallet public DID
+    if is_author_role(profile) and profile.context.settings.get_value(
+        "endorser.auto_promote_author_did"
+    ):
+        did = event.payload["did"]
+        try:
+            await promote_wallet_public_did(
+                profile, profile.context, profile.session, did
+            )
+        except Exception as e:
+            # log the error, but continue
+            LOGGER.exception(
```

Comment:
`exception` will produce a stack trace with the error being handled, it shouldn't need to be added to the message

Reply:
This is just writing the exception details to the logger, right?  It shouldn't affect any of the messaging.

Reply:
I mean the message argument that is passed to `exception`. There's a traceback added automatically so that should just be descriptive of the context. For example:

```
>>> import logging
>>> 
>>> try:
...     raise ValueError("nope")
... except:
...     logging.getLogger().exception("uh oh")
... 
uh oh
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: nope
```

Reply:
I found we have a bunch of cases of this at the moment which is counterproductive:

```
>>> import logging
>>> 
>>> try:
...     raise ValueError("nope")
... except Exception as err:
...     logging.getLogger().exception(err)
... 
nope
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: nope
```

Reply:
Ah ok I get it thanks

Reply:
Fixed